### PR TITLE
Promise.reject uses an Error instead of a string

### PR DESCRIPTION
**Pull request by theBadT**
Closes https://github.com/theia-ide/theia/issues/4560

### CODE REVIEW

File: packages/core/src/browser/opener-service.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ export class DefaultOpenerService implements OpenerService {
         if (handlers.length >= 1) {
             return handlers[0];
         }
-        return Promise.reject(`There is no opener for ${uri}.`);
+        return Promise.reject(Error('There is no opener for ${uri}.'));
```
Comment (svenefftinge) [inline packages/core/src/browser/opener-service.ts]: please keep the interpolation working

Comment (theBadT) [reply packages/core/src/browser/opener-service.ts]: my fault, will fix it

File: packages/bunyan/src/node/bunyan-logger-server.ts [hunk 2]
```diff
@@ -68,7 +68,7 @@ export class BunyanLoggerServer implements ILoggerServer {
      * documentation.  */
     child(name: string): Promise<void> {
         if (name.length === 0) {
-            return Promise.reject("Can't create a logger with an empty name.");
+            return Promise.reject(Error("Can't create a logger with an empty name."));
```
Comment (marechal-p) [inline packages/bunyan/src/node/bunyan-logger-server.ts]: Weirdly enough, just doing `Error(message)` seems to work, but the usual way of creating an error is by doing `new Error(message)`. Maybe change to that and see if other places also forgot the `new` keyword?

Comment (theBadT) [reply packages/bunyan/src/node/bunyan-logger-server.ts]: a single such usage could be find. Has been fixed

Comment (marechal-p) [reply packages/bunyan/src/node/bunyan-logger-server.ts]: Ah @kittaakos already saw that, take this as a reminder I guess :p

**Pull request merged**